src/lib/base/component-manager.js

```diff
@@ -64,6 +64,27 @@ const nonBubblingEventsMap = {
 // match & select "[eventType] [css selector]"
 const handlerMethodPattern = new RegExp(`^(${events.join('|')}|${Object.keys(nonBubblingEventsMap).join('|')}) (.*)`);
 
+/**
+ * Injects fallback mouse events for non-touch devices to events object.
+ * 
+ * @param {!Object} events
+ */
+function injectFallbackMouseEvents(events) {
+    const fallbackableEvents = ['touchend', 'tap'];
+
+    const _events = { ...events };
```

Comment:
Thank you for the pull request! I think we deliberately want to modify the original object. What was the reason for creating and returning a new object here?

Reply:
You're welcome! I want to contribute more in the future.

I usually prefer cloning the reference, mutating it locally, and returning it. Because I think it makes the code more predictable and it makes handling errors easier.

What is your opinion on this?

Reply:
I believe here it's a semantic feature that we're mutating the same object rather than a preference, as the name also suggests to "inject" properties. Also, even though it's a micro-optimization, `{ ...events }` is a costlier operation than _maybe_ adding a new property.